commit b226a826d83d66806fae20fc3518dace8b86bacb
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Fri Nov 22 12:42:42 2019 +0100

    mac80211: add a comment about monitor-to-dev injection
    
    Add a note with a use-case for the monitor-to-dev injection
    mechanism in mac80211, reported by Ben Greear.
    
    Change-Id: I6456997ef9bc40b24ede860b6ef2fed5af49cf44
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index db38be1b75fa..a53af8cd3756 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -2270,6 +2270,9 @@ netdev_tx_t ieee80211_monitor_start_xmit(struct sk_buff *skb,
 	 * isn't always enough to find the interface to use; for proper
 	 * VLAN/WDS support we will need a different mechanism (which
 	 * likely isn't going to be monitor interfaces).
+	 *
+	 * This is necessary, for example, for old hostapd versions that
+	 * don't use nl80211-based management TX/RX.
 	 */
 	sdata = IEEE80211_DEV_TO_SUB_IF(dev);